src/Build/Evaluation/Expander.cs

```diff
@@ -4242,6 +4240,47 @@ private static bool TryGetArg(object[] args, out string arg0)
                 return arg0 != null;
             }
 
+            private static bool TryGetArgs(object[] args, out string arg0, out StringComparison arg1)
+            {
+                if (args.Length != 2)
+                {
+                    arg0 = null;
+                    arg1 = default;
+
+                    return false;
+                }
+
+                arg0 = args[0] as string;
+
+                // reject enums as ints. In C# this would require a cast, which is not supported in msbuild expressions
+                if (arg0 == null || !(args[1] is string comparisonTypeName) || ContainsDigits(comparisonTypeName))
+                {
+                    arg1 = default;
+                    return false;
+                }
+
+                // Allow fully-qualified enum, e.g. "System.StringComparison.OrdinalIgnoreCase"
+                if (comparisonTypeName.Contains('.'))
+                {
+                    comparisonTypeName = comparisonTypeName.Replace("System.StringComparison.", "").Replace("StringComparison.", "");
+                }
+
+                return Enum.TryParse(comparisonTypeName, out arg1);
+
+                bool ContainsDigits(string aString)
```

Comment:
Instead of writing this yourself, you could use:
`comparisonTypeName.Any(c => char.IsDigit(c))` above.

Reply:
While not a super hot path, it's still called pretty often so I wanted to avoid linq.

Reply:
The Linq version is indeed significantly slower. I understand the intention here is to see if the argument is a number as opposed to an identifier. Noting that identifiers may have digits in them (luckily not `StringComparison` fields, at least in its current version), I'm wondering if `int.TryParse` wouldn't be a more robust check. And if not, maybe it would make sense to move this helper to Shared\StringExtensions.cs.

Reply:
I like using `int.TryParse`. Does that work?

Reply:
Yup, changed it.